src/ArrowFunctions.h

```diff
@@ -3,13 +3,18 @@
 
 // Wrap functions in C extern if compiling C++ object file
 #ifdef __cplusplus
+#include <cassert>
+#include <fstream>
 #include <iostream>
+#include <memory>
 #include <arrow/api.h>
 #include <arrow/io/api.h>
+#include <arrow/table.h>
```

Comment:
are we using `<arrow/table.h>` anywhere? It would be nice to double check all the new imports to make sure they're still needed

Reply:
That's odd, table should not be new, that was being used by the append code (which was not removed). Let me look into this.

Reply:
I think I added some of these when trying out a few alternative methods, if memory serves correctly only `queue` is needed. I will remove and retest to be positive.

Reply:
Verified only queue is needed. Updated

Reply:
sweet!